_sips/minutes/2017-10-24-sip-minutes.md

```diff
@@ -0,0 +1,233 @@
+| f"" interpolator   |  x             |            |        |         |
+| Json.serialize\[T] |                | x          |        |         |
+| ScalaTest Position |                | x          |        |         |
+| sourcecode.Name    |                | x          |        |         |
+| log4s logger.info  |                |            | x      |         |
+| Refined Positive   |                |            |        | x       |
+
+We can view these categories by the features they require:
+
+|                       | transformation | generation | inline | linting |
+| -------------------   | -------------- | ---------- | ------ | ------- |
+| Inspect trees         | x              |            | x      | x       |
+| Inspect types/symbols | x              | x          |        |         |
+| Construct trees       | x              | x          |        |         |
+| Report errors         | x              | x          | x      | x       |
+| Solvable with SIP-28  |                |            | x      |         |
+
+
+*Transformation macros*
+
+There are still many hard/open/unsolved problems, most notably:
+* splicing untyped tree under typed trees causes breaks typer invariants, causing compiler crashes
+* splicing typed tree under untyped trees causes owner chain corruptions, causing compiler crashes
+
+Several techniques have been explored to solve these problems:
+
+* c.untypecheck, breaks for certain tree nodes due to non-idempotency
+* c.typecheck or manually construct typed trees, requires expertise
+  from macro authors and can still cause cryptic errors in later phases.
+* automatic repair of owner chains, works for some limited macros
+* automatic typechecking of spliced untyped trees, little explored
+
+These solutions have different trade-offs with regards to
+
+- feature support
+- portability
+- breaking changes with existing macro ecosystem
+
+
+*Generation macros*
+
+There seem to be no roadblockers for supporting macros that do no tree
+inspection, only untyped tree construction.
+Code generation macros cannot change.
+
+Open discussion about the above proposed
+
+[YouTube time: 23'39'' until the end](https://youtu.be/aIc-o1pcRhw?t=1419)
+
+About 40 minutes were dedicated to finding a common ground for the  direction of the project as well as the technical details that should be addressed going forward.
+
+**Adriaan** immediately pointed out that by taking on the "hard stuff" to deal with first will give clarity to what could be supported for the next year. He suggests tackling the more ambitious features first, by experimenting in a current macro system, is a "cheap way" to find out the "unknown". The goal would be to find out how to implement it in the next macro system.
+Challenges that can be taken on:
+[(https://youtu.be/aIc-o1pcRhw?t=1673)]
+- splicing trees, ending up with owner chains that are correct
+- hygiene
+- better tools for macro authors, to experiment in a current macro system
+
```

Comment:
@adriaanm please take a look and let me know if I got all you wanted to come across, thanks!

Reply:
Thanks, reads better than I said it :-)

Reply:
@adriaanm ohhh, you are being too nice (o: I re-wrote a bit, so it's even better hihihi